cli/src/utils.rs

```diff
@@ -41,6 +39,31 @@ pub fn format_devices(
     Ok(())
 }
 
+pub fn get_minimal_short_id_size<'a, I>(devices: I) -> usize
+where
+    I: Iterator<Item = &'a libparsec::DeviceID>,
+{
+    use itertools::Itertools;
+
+    devices
+        // Ensure we do not have duplicates in the set.
+        .sorted_by(|a, b| a.cmp(b))
```

Comment:
I don't think dedup is needed here:

The list provided by the caller is expected to be printed with the IDs shortened according to what we return.
So in case of duplication we do want to display the full IDs an not a subset, otherwise we will end up with multiple devices with the same short IDs, making us believe we do have multiple devices available instead of realizing there is a bug in the code !

Reply:
> So in case of duplication we do want to display the full IDs an not a subset, otherwise we will end up with multiple devices with the same short IDs

The algorithm already handle that case, see the test case where only the last hex char is different.